src/Jellyfin.XmlTv/XmlTvReader.cs

```diff
@@ -8,1104 +6,1342 @@
-            if (string.IsNullOrEmpty(endValue))
-            {
-                result.EndDate = DateTimeOffset.MinValue;
-            }
-            else
-            {
-                result.EndDate = ParseDate(endValue).GetValueOrDefault();
-            }
+            setter(value);
         }
+    }
 
-        public static DateTimeOffset? ParseDate(string dateValue)
+    private void PopulateHeader(XmlReader reader, XmlTvProgram result)
+    {
+        var startValue = reader.GetAttribute("start");
+        if (string.IsNullOrEmpty(startValue))
         {
-            /*
-            All dates and times in this DTD follow the same format, loosely based
-            on ISO 8601.  They can be 'YYYYMMDDhhmmss' or some initial
-            substring, for example if you only know the year and month you can
-            have 'YYYYMM'.  You can also append a timezone to the end; if no
-            explicit timezone is given, UTC is assumed.  Examples:
-            '200007281733 BST', '200209', '19880523083000 +0300'.  (BST == +0100.)
-            */
-
-            if (string.IsNullOrEmpty(dateValue))
-            {
-                return null;
-            }
+            result.StartDate = DateTimeOffset.MinValue;
+        }
+        else
+        {
+            result.StartDate = ParseDate(startValue).GetValueOrDefault();
+        }
 
-            const string CompleteDate = "20000101000000";
-            var dateComponent = string.Empty;
-            string? dateOffset = null;
-            var match = Regex.Match(dateValue, DateWithOffsetRegex);
-            if (match.Success)
-            {
-                dateComponent = match.Groups["dateDigits"].Value;
-                if (!string.IsNullOrEmpty(match.Groups["dateOffset"].Value))
-                {
-                    var tmpDateOffset = match.Groups["dateOffset"].Value; // Add in the colon to ease parsing later
-                    if (tmpDateOffset.Length == 5)
-                    {
-                        dateOffset = tmpDateOffset.Insert(3, ":"); // Add in the colon to ease parsing later
-                    }
-                }
-            }
-            else
-            {
-                return null;
-            }
+        var endValue = reader.GetAttribute("stop");
+        if (string.IsNullOrEmpty(endValue))
+        {
+            result.EndDate = DateTimeOffset.MinValue;
+        }
+        else
+        {
+            result.EndDate = ParseDate(endValue).GetValueOrDefault();
+        }
+    }
 
-            // Pad out the date component part to 14 characaters so 2016061509 becomes 20160615090000
-            if (dateComponent.Length < 14)
-            {
-                dateComponent += CompleteDate[dateComponent.Length..];
-            }
+    /// <summary>
+    /// Parses a string into <see cref="DateTimeOffset"/> if possible.
+    /// </summary>
+    /// <param name="dateValue">The input string.</param>
+    /// <returns>DateTimeOffset.</returns>
```

Comment:
```suggestion
    /// <returns>A <see cref="DateTimeOffset"/> if the <c>dateValue</c> parameter was converted successfully; otherwise, <c>null</c>.</returns>
```